admin/modules/forum/attachments.php

```diff
@@ -858,14 +858,21 @@ function() {
 			if($num_results > $mybb->input['perpage'])
 			{
 				$pagination_url = "index.php?module=forum-attachments&amp;results=1";
-				$pagination_vars = array('perpage', 'sortby', 'order', 'filename', 'mimetype', 'username', 'fid', 'downloads', 'downloads_dir', 'dateuploaded', 'dateuploaded_dir', 'filesize', 'filesize_dir');
+				$pagination_vars = array('perpage', 'sortby', 'order', 'filename', 'mimetype', 'username', 'downloads', 'downloads_dir', 'dateuploaded', 'dateuploaded_dir', 'filesize', 'filesize_dir');
 				foreach($pagination_vars as $var)
 				{
 					if($mybb->input[$var])
 					{
 						$pagination_url .= "&{$var}=".urlencode($mybb->input[$var]);
 					}
 				}
+				if($mybb->input['forum'])
```

Comment:
Might be good to change to this:

```suggestion
				if(is_array($mybb->input['forum']) && !empty($mybb->input['forum'])
```

Reply:
<del>
If we use `is_array()` on `$mybb->input['forum']`, we must ensure `$mybb->input['forum']` exists, thus an `isset()` should be called before `is_array()`. We will get an `Undefined variable` notice with your modification, if no forum is selected in the query form, I suppose.

Don't know how to make a change to your change, so I leave a comment. Hope you have time to check it, @WildcardSearch.
</del>

Oh, I've just seen that MyBB suppresses NOTICE error types. This code works!